### PR TITLE
fix(onboard): roll back orphaned sandbox when dashboard alloc fails (#2174 followup)

### PR DESCRIPTION
**Pull request by cjagwani**
## Summary

Followup to #2411 (merged). Closes the second half of #2174's title — "leaks ghost sandbox" — that the merged fix didn't address.

When `findAvailableDashboardPort` exhausts the port range it `throw`s. By the time `ensureDashboardForward` runs (`src/lib/onboard.ts:3830`), the openshell sandbox has already been streamed into existence (`streamSandboxCreate` at line 3734, ready-wait at lines 3789–3803). The throw escapes unhandled, leaving the sandbox in `openshell sandbox list` with no NemoClaw registry entry — exactly the drift reported in #2174.

Issue comment with the analysis: https://github.com/NVIDIA/NemoClaw/issues/2174#issuecomment-4336937684

## Related Issue

Followup to #2174 (closed by #2411). #2411 fixed the crash; this PR fixes the leak.

## Changes

- `src/lib/onboard.ts` — `ensureDashboardForward` takes a new `{ rollbackSandboxOnFailure?: boolean }` option. On unrecoverable port-allocation failure (range exhaustion), the just-created openshell sandbox is deleted, an actionable error is printed, and the process exits with code 1 — mirroring the not-ready rollback pattern already in place at lines 3789–3803.
- The post-create call site (line 3830) opts in (`rollbackSandboxOnFailure: true`).
- The four reuse-path call sites (lines 3295, 3326, 3372, 3390) keep the default (`false`) — they operate on already-existing sandboxes where deleting on alloc failure would destroy user state.

## Type of Change

- [x] Code change (feature, bug fix, or refactor)

## Verification

- [x] `npm run build:cli` + `npm run typecheck:cli` green
- [x] `npx vitest run --project cli` — 2250/2250 passed
- [x] `npx vitest run --project plugin` — 410/410 passed
- [x] `npx prek run --files src/lib/onboard.ts` — green
- [x] No new tests added: the rollback path mirrors the existing not-ready rollback (3789–3803), which also has no dedicated unit test. Adding test infrastructure here would exceed the minimum scope of this followup.
- [x] No secrets, API keys, or credentials committed

## Notes for reviewer

- Diff is ~35 lines on a single file. Surgical fix matching an existing pattern in the same function.
- Doc comment on `ensureDashboardForward` updated to describe the new option and reference the pattern it mirrors.

## AI Disclosure

- [x] AI-assisted — tool: Claude Code

Signed-off-by: Charan Jagwani <cjagwani@nvidia.com>

## Summary by CodeRabbit

* **Bug Fixes**
  * Sandbox initialization now attempts automatic rollback if dashboard port allocation fails, and exits safely when rollback succeeds.
  * When automatic cleanup fails, users are shown clear manual-cleanup instructions and the exact command to remove the orphaned sandbox.

* **Tests**
  * Added runtime tests verifying rollback messaging covers both successful and failed cleanup scenarios and accurate sandbox naming in instructions.

### CODE REVIEW

File: test/onboard-rollback.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+function isOnboardRollbackInternals(value: object | null): value is OnboardRollbackInternals {
+  return (
+    value !== null &&
+    typeof Reflect.get(value, "buildOrphanedSandboxRollbackMessage") === "function"
+  );
+}
+
+const loadedOnboardInternals = require("../dist/lib/onboard");
```
[the remark this answers is not in the document]
Comment (cjagwani) [reply test/onboard-rollback.test.ts:27]: Pushing back on this. Tried the conversion in commit ef9b0d5e then reverted.

Adding `export` to `buildOrphanedSandboxRollbackMessage` in `src/lib/onboard.ts` (so the named ESM import would type-check) reshapes `dist/lib/onboard.d.ts` from `unknown`-shaped to a typed module. That breaks two other tests:

- `test/ssh-known-hosts.test.ts:27` — TS2345
- `test/wsl2-probe-timeout.test.ts:26,29` — TS2559 + TS2345

Both rely on the `d.ts` staying opaque so their runtime guards (`isOnboardKnownHostsInternals`, `isOnboardValidationCandidate`) type-narrow against `OnboardX | typeof import("../dist/lib/onboard") | null`. Once `import("../dist/lib/onboard")` has known properties, TS rejects "no properties in common" with the `OnboardXCandidate` shape and the call sites stop compiling.

Keeping `require()` here matches the convention already used for this same file by:
- `test/credential-rotation.test.ts:43`
- `test/gateway-start-wait.test.ts:15`
- `test/gemini-probe-auth.test.ts:14`
- `test/ssh-known-hosts.test.ts`
- `test/wsl2-probe-timeout.test.ts`

The mixed pattern in `test/` reflects whether the imported module uses per-function `export` (pure-function modules like `policies`, `runner`, `resolve-openshell`) or bottom-of-file `module.exports = {...}` with runtime-guarded internals (`onboard`). The CLAUDE.md ESM rule applies cleanly to the former; the latter would need a coordinated sweep of `onboard.ts` + the four neighbouring tests, which is out of scope for #2174's leak fix.

Documented this rationale inline in `test/onboard-rollback.test.ts` so the next reader doesn't re-litigate.

The other CR finding (JSDoc alignment) is fixed in the same commit.

**Review by cv — APPROVED** [review]
